modules/edit-modes/src/lib/rotate-mode.js

```diff
@@ -1,84 +1,195 @@
 // @flow
 
+import turfBuffer from '@turf/buffer';
+import bbox from '@turf/bbox';
 import turfCentroid from '@turf/centroid';
 import turfBearing from '@turf/bearing';
+import bboxPolygon from '@turf/bbox-polygon';
+import turfDistance from '@turf/distance';
+import { coordEach } from '@turf/meta';
+import { getGeom } from '@turf/invariant';
+import { point, featureCollection, lineString } from '@turf/helpers';
 import turfTransformRotate from '@turf/transform-rotate';
 import type {
   PointerMoveEvent,
   StartDraggingEvent,
   StopDraggingEvent,
   DraggingEvent,
-  ModeProps
+  ModeProps,
+  EditHandleFeature
 } from '../types.js';
+import { getPickedEditHandle } from '../utils';
 import type { FeatureCollection, Position } from '../geojson-types.js';
-import { BaseGeoJsonEditMode, type GeoJsonEditAction } from './geojson-edit-mode.js';
+import {
+  BaseGeoJsonEditMode,
+  type GeoJsonEditAction,
+  getIntermediatePosition
+} from './geojson-edit-mode.js';
 import { ImmutableFeatureCollection } from './immutable-feature-collection.js';
 
 export class RotateMode extends BaseGeoJsonEditMode {
-  _isRotatable: boolean;
-  _geometryBeingRotated: ?FeatureCollection;
+  _selectedEditHandle: ?EditHandleFeature;
+  _selectedGeometry: ?FeatureCollection;
+  _previousMouseCoords: ?Position;
+  _isRotating: boolean = false;
+  _isCompositionMode: boolean;
+
+  constructor(options: ?{ isCompositionMode: boolean }) {
+    super();
+    const { isCompositionMode } = options || {};
+    this._isCompositionMode = isCompositionMode || false;
+  }
+
+  _isSinglePointGeometrySelected = (): boolean => {
+    const { features } = this._selectedGeometry || {};
+    if (Array.isArray(features) && features.length === 1) {
+      const { type } = getGeom(features[0]);
+      return type === 'Point';
+    }
+    return false;
+  };
+
+  getIsRotating = () => this._isRotating;
+
+  getGuides(props: ModeProps<FeatureCollection>) {
+    this._selectedGeometry = this.getSelectedFeaturesAsFeatureCollection(props);
```

Comment:
I think we should tweak this so that we capture the selected geometry before rotation (in `handleStartDragging` as it was before). This way, rounding errors don't accumulate. This happens when you rotate something, then rotate the rotated geometry, then repeat over and over on mouse move.

See this demo of the rounding errors accumulating. Notice how I start dragging and stop dragging at the top, yet the geometry is not close to its original rotation.
![rotate-bug](https://user-images.githubusercontent.com/469582/76436503-ef980280-637d-11ea-8585-e0184052d94d.gif)